packages/webapp/src/components/Animals/RemoveAnimalsModal/index.tsx

```diff
@@ -0,0 +1,196 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import Drawer from '../../Drawer';
+import Button from '../../Form/Button';
+import ReactSelect from '../../Form/ReactSelect';
+import styles from './styles.module.scss';
+import { Controller, useForm, SubmitHandler } from 'react-hook-form';
+import { useTheme, useMediaQuery } from '@mui/material';
+import clsx from 'clsx';
+import Input from '../../Form/Input';
+import { ReactComponent as WarningIcon } from '../../../assets/images/warning.svg';
+import { ReactComponent as CheckIcon } from '../../../assets/images/check-circle.svg';
+import { useTranslation } from 'react-i18next';
+import { ClassNamesConfig } from 'react-select';
+
+const REASON = 'reason';
+const EXPLANATION = 'explanation';
+
+type FormFields = {
+  [REASON]: string;
+  [EXPLANATION]: string;
+};
+
+enum RemovalOptionValue {
+  SOLD = 'SOLD',
+  SLAUGHTERED_FOR_SALE = 'SLAUGHTERED_FOR_SALE',
+  SLAUGHTERED_FOR_CONSUMPTION = 'SLAUGHTERED_FOR_CONSUMPTION',
+  NATURAL_DEATH = 'NATURAL_DEATH',
+  CULLED = 'CULLED',
+  OTHER = 'OTHER',
+  CREATED_IN_ERROR = 'CREATED_IN_ERROR',
+}
+
+type RemovalOption = { label: string; value: RemovalOptionValue };
+
+type RemoveAnimalsModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  onConfirm: SubmitHandler<FormFields>;
+  showSuccessMessage: boolean;
+};
+
+export default function RemoveAnimalsModal(props: RemoveAnimalsModalProps) {
+  const { register, handleSubmit, control, watch } = useForm<FormFields>();
+  const { t } = useTranslation();
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
```

Comment:
I'm wondering if this can be better used as a custom hook? 
```

function useIsMobile() {
    const theme = useTheme();
    const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
    return isMobile
}
```

Reply:
We could, although I think we're not always necessarily just going to have one breakpoint so we'd want to keep the breakpoints generic enough